### PR TITLE
DOCS Document deprecated API and a few new features

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/11341

### CODE REVIEW

File: en/08_Changelogs/5.4.0.md [hunk 1]
```diff
@@ -15,16 +15,57 @@ title: 5.4.0 (unreleased)
...
+- A new [`BaseKernel::getBooted()`](api:SilverStripe\Core\BaseKernel::getBooted()) method has been added for checking whether the kernel has been booted yet or not.
+- A new [`CoreKernel::setBootDatabase()`](api:SilverStripe\Core\CoreKernel::setBootDatabase()) method has been added to replace the now deprecated [`DatabaselessKernel`](api:SilverStripe\Core\DatabaselessKernel) class.
+- Two new methods have been added to [`ArrayLib`](api:SilverStripe\ORM\ArrayLib):
+  - [`ArrayLib::insertBefore()`](api:SilverStripe\ORM\ArrayLib::insertBefore()) for inserting a value before another value in an array
+  - [`ArrayLib::insertAfter()`](api:SilverStripe\ORM\ArrayLib::insertAfter()) for inserting a value after another value in an array
+- A new [`DBDatetime::getTimeBetween()`](api:SilverStripe\ORM\FieldType\DBDatetime::getTimeBetween()) method has been added. This method returns the amount of time that has passed between two `DBDateTime` objects as a human-readable string.
+- A new [`AbstractQueuedJob::getQueue()`](api:Symbiote\QueuedJobs\Services\AbstractQueuedJob::getQueue()) static method has been added to get the correct queue constant from a given string or int.
```
Comment (emteknetnz) [inline en/08_Changelogs/5.4.0.md:43]: May need to change this line if updating method name

File: en/08_Changelogs/5.4.0.md [hunk 2]
```diff
@@ -15,16 +15,57 @@ title: 5.4.0 (unreleased)
...
+- [`DatabaselessKernel`](api:SilverStripe\Core\DatabaselessKernel) has been deprecated. Use [`CoreKernel::setBootDatabase()`](api:SilverStripe\Core\CoreKernel::setBootDatabase()) instead.
+- [`BuildTask.segment`](api:SilverStripe\Dev\BuildTask->segment) has been deprecated. It will be replaced with a new `$commandName` property.
+- [`BuildTask->description`](api:SilverStripe\Dev\BuildTask->description) has been deprecated. It will be replaced with a new static property with the same name.
+- [`BuildTask::getDescription()`](api:SilverStripe\Dev\BuildTask::getDescription()) has been deprecated. It will be replaced with a new static method with the same name.
+- [`DevBuildController`](api:SilverStripe\Dev\DevBuildController) has been deprecated. It will be replaced with a new `SilverStripe\Dev\Command\DbBuild` class.
+- [`DevConfigController`](api:SilverStripe\Dev\DevConfigController) has been deprecated. It will be replaced with a new `SilverStripe\Dev\Command\ConfigDump` class.
+- [`DatabaseAdminDatabaseAdmin`](api:SilverStripe\ORM\DatabaseAdminDatabaseAdmin) has been deprecated. It will be replaced with a new `SilverStripe\Dev\Command\DbBuild` class.
```
Comment (emteknetnz) [inline en/08_Changelogs/5.4.0.md]: I think this is supposed to simply be `DatabaseAdmin`

File: en/08_Changelogs/5.4.0.md [hunk 3]
```diff
@@ -15,16 +15,57 @@ title: 5.4.0 (unreleased)
...
+- [`DevelopmentAdmin::buildDefaults()`](api:SilverStripe\Dev\DevelopmentAdmin::buildDefaults()) has been deprecated. It will be replaced with a new `SilverStripe\Dev\Command\DbDefaults` class.
+- [`DevelopmentAdmin::generatesecuretoken()`](api:SilverStripe\Dev\DevelopmentAdmin::generatesecuretoken()) has been deprecated. It will be replaced with a new `SilverStripe\Dev\Command\GenerateSecureToken` class.
+- [`DevelopmentAdmin::getRegisteredController()`](api:SilverStripe\Dev\DevelopmentAdmin::getRegisteredController()) has been deprecated. It will be removed without equivalent functionality to replace it.
+- [`DevelopmentAdmin.registered_controllers`](api:SilverStripe\Dev\DevelopmentAdmin->registered_controllers) has been deprecated. It will be replaced with new `controllers` and `commands` configuration properties.
+- [`CleanupTestDatabasesTask::canView()`](api:SilverStripe\Dev\Tasks\CleanupTestDatabasesTask::canView()) has been deprecated. It will be replaced with a new `canRunInBrowser()` method.
+- [`HTTPOutputHandler`](api:SilverStripe\Logging\HTTPOutputHandler) has been deprecated. It will be renamed to `SilverStripe\Logging\ErrorOutputHandler`
```
Comment (emteknetnz) [inline en/08_Changelogs/5.4.0.md:78]: A little weird that you also have HTTPOutputHandler.isCli() deprecated, which has a replacement - either don't deprecate it or mention it here

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: en/08_Changelogs/5.4.0.md (hunk 3 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/5.4.0.md:78]: There's a lot of deprecations here, so forgetting one is understandable I think lol.
Added.

File: en/08_Changelogs/5.4.0.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/5.4.0.md]: yep good catch. Fixed.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**